src/videojs-contrib-hls.js

```diff
@@ -87,6 +87,9 @@ export default class HlsHandler extends Component {
     // HlsSourceHandler.handleSource()
     // @link https://github.com/videojs/video.js/pull/3245
     this.options_ = videojs.mergeOptions(options, videojs.options.hls || {});
+    // we cannot override the mode at this state, we have to use what was passed
+    // to us
+    this.options_.mode = options.mode;
```

Comment:
Well if we get here with a with `videojs.options.hls.mode = 'flash'` but the tech is currently html and we were passed `mode = 'html'` by the HlsSourceHandler a MediaSource error will be thrown, so we only ever want to use what HlsSourceHandler passes us.